### DISCUSSION TITLE
An agent framework leveraging Spring AI

### DISCUSSION THREAD

**Discussion by thesurlydev** [spring-projects/spring-ai#612] [Show and tell]
I've been at work on an Agent framework that leverages Spring AI here:

https://github.com/thesurlydev/spring-ai-collab

The intent was to build an agent framework that caters to Spring users. It currently has several example Agents and tools which use annotations to describe them. It leverages the LLM to determine which agent and tool to use to accomplish tasks.

Note that the tool support is independent of any function calling support by LLMs. This was a design choice to side step the fact that some LLMs don't support function calling or implement it differently.

Some of the planned features I'm holding off on because they'll depend on chat message history which I believe is still in the works.

I'd appreciate any feedback and I'm happy to answer any questions.

**Comment by bibekditagtech** [comment]
Huge respect, I have just finish checking out all the feature related to agents. it is quite awesome. I really like this https://prnt.sc/7F1OB1LPaMnP

**Comment by kdhrubo** [comment]
Can we not build this kind of systems using an workflow engine?
